src/resmom/requests.c

```diff
@@ -1543,9 +1543,11 @@ post_suspend(job *pjob, int err)
 		stop_walltime(pjob);
 
 		pjob->ji_polltime = 0;	/* don't check polling */
-		pjob->ji_qs.ji_substate = JOB_SUBSTATE_SUSPEND;
-		pjob->ji_qs.ji_svrflags |= JOB_SVFLG_Suspend;
-		(void)job_save(pjob, SAVEJOB_QUICK);
+		if (pjob->ji_qs.ji_substate < JOB_SUBSTATE_EXITING) {
```

Comment:
Can you log a DEBUG message that why a job is not moving to suspend sub-state? Fix looks good otherwise.

Reply:
I agree with @arungrover - i would even say that it is better to add a info level log (so it appears even in low log levels) to log that we are not changing the substate since the job was in exiting or higher state (and state the substate that we did not change to)

Otherwise the code change looks good

Reply:
Thanks, @arungrover & @subhasisb for your review comments 
I have added log_event for INFO level. Please have a look and share your comments/sign-off.